src/main/java/com/rad/multiplex/controller/MultiplexController.java

```diff
@@ -0,0 +1,85 @@
+package com.rad.multiplex.controller;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.time.LocalDateTime;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.reflect.TypeToken;
+import com.rad.multiplex.model.BadRequestException;
+import com.rad.multiplex.model.LocalDateTimeConverter;
+import com.rad.multiplex.model.Reservation;
+import com.rad.multiplex.model.Response;
+import com.rad.multiplex.model.Screening;
+import com.rad.multiplex.service.ScreeningService;
+
+@RestController
+public class MultiplexController {
+
+	@Autowired
+	private ScreeningService screeningService;
+
+	private Gson gson = new GsonBuilder()
+			.excludeFieldsWithoutExposeAnnotation()
+			.registerTypeAdapter(LocalDateTime.class, new LocalDateTimeConverter()).create();
+
+	@PostMapping("/interval")
+	public String getInterval(@RequestBody String string)
+			throws JsonSyntaxException, UnsupportedEncodingException {
+
+		string = URLDecoder.decode(string, "UTF-8");
+
+		List<LocalDateTime> times =gson.fromJson(string, new TypeToken<List<LocalDateTime>>() {
+				}.getType());
+
+		if (times.size()!=2) {
+			throw new BadRequestException("malformed request");
+		}
+		
+		List<Screening> byInterval = screeningService.getInterval(times.get(0), times.get(1));
+	
+		return gson.toJson(byInterval);
+
+	}
+
+	@PostMapping("/getseats")

```

Comment:
Nazwy URL nie są zbyt REST-owe. Poza tym get POST-em?

Reply:
Obiło mi się o uszy, że panuje dość duża dowolność w tym względzie, tzn. że można GET zrobić POSTEM i że nie jest to zła praktyka. Poza tym POST ma większą przepustowość... A co do nazw, faktycznie, dość roboczo brzmią niestety.